rts/Sim/Units/CommandAI/MobileCAI.cpp

```diff
@@ -1018,35 +1022,75 @@ void CMobileCAI::NonMoving()
 	if (lastBuggerOffTime <= (gs->frameNum - BUGGER_OFF_TTL))
 		return;
 
-	if (((owner->pos - buggerOffPos) * XZVector).SqLength() >= Square(buggerOffRadius * 1.5f))
+	// increase the target distance if continuing to fail to clear
+	float targetDistance = buggerOffRadius + owner->radius*(1 + 0.3*buggerOffAttempts);
+	if (((owner->pos - buggerOffPos) * XZVector).SqLength() >= Square(targetDistance))
 		return;
 
-	float3 buggerVec;
+	float3 buggerVec = ZeroVector;
```

Comment:
This should be covered by the default constructor of `float3`

Reply:
:+1: Removed the zerovector assignments.